homeassistant/components/xiaomi_miio/fan.py

```diff
@@ -736,7 +797,10 @@ def speed_list(self) -> list:
     def speed(self):
         """Return the current speed."""
         if self._state:
-            return AirpurifierOperationMode(self._state_attrs[ATTR_MODE]).name
+            if isinstance(self._device, AirPurifierMiot):
+                return AirpurifierMiotOperationMode(self._state_attrs[ATTR_MODE]).name
```

Comment:
Would it make more sense to save a reference to the mode as given by the library and simply use `_speed.name`? Usage of `isinstance` is generally speaking not a good thing.

Reply:
I've changed this to get rid of runtime instance checking